userapi/internal/user_api.go

```diff
@@ -386,11 +386,6 @@ func (a *UserInternalAPI) PerformDeviceUpdate(ctx context.Context, req *api.Perf
 	}
 	res.DeviceExists = true
 
-	if dev.UserID != req.RequestingUserID {
```

Comment:
Removed because we should never ever hit this case, because we are querying the database for the `dev.UserID`. (except the database is lying to us?)

Reply:
We need to look up the device, as one device can update another of the same user.